modules/microvm/virtualization/microvm/adminvm.nix

```diff
@@ -11,10 +11,14 @@
 
   adminvmBaseConfiguration = {
     imports = [
-      (import ./common/vm-networking.nix {inherit vmName macAddress;})
+      (import ./common/vm-networking.nix {
+        inherit config lib vmName macAddress;
+        internalIP = 10;
```

Comment:
For all of the places where the internalIP is set should this not come from the modules/common/networking/hosts.nix? like this we are doing the accounting twice. or is this coming in a followup patch?

Reply:
Yes, removal of IP duplication is to come in a follow-up patch. We actually tried that first but with a wrong approach. Merging of VMs declared IPs with built-time generated /etc/hosts that go into the VMs turned out as a build-time chicken-egg problem with nix - even with nix expert support.

I'd rather not debate that further in this PR but [refactor that in the follow-up PR like this](https://github.com/tiiuae/ghaf/pull/584#discussion_r1628929192).